idflakies-maven-plugin/src/main/java/edu/illinois/cs/dt/tools/plugin/IncDetectorMojo.java

```diff
@@ -0,0 +1,651 @@
+        if (!new File(oldSfPathFileName).exists()) {
+            return false;
+        }
+        try {
+            List<String> oldClassPathLines = Files.readAllLines(Paths.get(oldSfPathFileName));
+            if (oldClassPathLines.size() != 1) {
+                throw new MojoExecutionException(SF_CLASSPATH + " is corrupt! Expected only 1 line.");
+                // This exception is not correct and need to be modified.
+            }
+            List<String> oldClassPathelements = getCleanClassPath(oldClassPathLines.get(0));
+            // comparing lists and not sets in case order changes
+            if (sfPathString.equals(oldClassPathelements)) {
+                return true;
+            }
+            Set<String> sfPathStringSet = new HashSet<>();
+            sfPathStringSet.addAll(sfPathString);
+            Set<String> oldClassPathelementsSet = new HashSet<>();
+            oldClassPathelementsSet.addAll(sfPathString);
+            if (sfPathStringSet.equals(oldClassPathelementsSet)) {
+                return true;
+            }
+        } catch (IOException ioe) {
+            ioe.printStackTrace();
+        }
+        return false;
+    }
+
+    private static Set<String> getImmutableList() {
+        if (immutableList == null) {
+            immutableList = new HashSet<>();
+
+            immutableList.add("java.lang.String");
+            immutableList.add("java.lang.Enum");
+            immutableList.add("java.lang.StackTraceElement");
+            immutableList.add("java.math.BigInteger");
+            immutableList.add("java.math.BigDecimal");
+            immutableList.add("java.io.File");
+            immutableList.add("java.awt.Font");
+            immutableList.add("java.awt.BasicStroke");
+            immutableList.add("java.awt.Color");
+            immutableList.add("java.awt.GradientPaint");
+            immutableList.add("java.awt.LinearGradientPaint");
+            immutableList.add("java.awt.RadialGradientPaint");
+            immutableList.add("java.awt.Cursor");
+            immutableList.add("java.util.Locale");
+            immutableList.add("java.util.UUID");
+            immutableList.add("java.util.Collections");
+            immutableList.add("java.net.URL");
+            immutableList.add("java.net.URI");
+            immutableList.add("java.net.Inet4Address");
+            immutableList.add("java.net.Inet6Address");
+            immutableList.add("java.net.InetSocketAddress");
+            immutableList.add("java.awt.BasicStroke");
+            immutableList.add("java.awt.Color");
+            immutableList.add("java.awt.GradientPaint");
+            immutableList.add("java.awt.LinearGradientPaint");
+            immutableList.add("java.awt.RadialGradientPaint");
+            immutableList.add("java.awt.Cursor");
+            immutableList.add("java.util.regex.Pattern");
+        }
+        return immutableList;
+    }
+
+    private static boolean inImmutableList(Field field) {
```

Comment:
Technically, this method is just checking whether field is immutable and final, not just that it is in that immutable list; maybe can just call it `isImmutable`.